internal/job/shell/olfactory_error.go

```diff
@@ -34,8 +31,10 @@ func (e *OlfactoryError) Unwrap() error {
 	return e.inner
 }
 
-// Is returns true if the target is an OlfactoryError and the inner errors are equal
+// Is returns true if the target is an OlfactoryError and the smells match. If
+// you want to check the inner errors match, you should unwrap them and run
+// errors.Is on the unwrapped errors.
 func (e *OlfactoryError) Is(target error) bool {
 	terr, ok := target.(*OlfactoryError)
-	return ok && e.Smell == terr.Smell && errors.Is(e.inner, terr.inner)
```

Comment:
I was a bit overzealous with error unwrapping here. I think in most cases, we don't care what the inner errors are. If we do, there's a way to figure it out.